imapautofiler/rules.py

```diff
@@ -254,6 +255,28 @@ def __init__(self, rule_data, cfg):
         super().__init__(rule_data, cfg)
 
 
+class TimeLimit(Rule):
+    """True if message is older than the specified 'age' measured
+     in number of days."""
+
+    _log = logging.getLogger('TimeLimit')
+    NAME = 'time-limit'
+
+    def __init__(self, rule_data, cfg, ):
+        super().__init__(rule_data, cfg)
+        self._age = rule_data['time-limit']['age']
+
+    def check(self, message):
+        date = parsedate_to_datetime(message['date'])
```

Comment:
It looks like the test is failing because message['date'] is sometimes None. I think we'll need to add an explicit check for that case. If the date is None or an empty string the check should fail.